ams/AssetAnalyzer.cs

```diff
@@ -30,24 +32,48 @@ internal class AssetAnalyzer : BaseMigrator
             _tracker = tracker;
         }
 
-        private async Task<AnalysisResult> AnalyzeAsync(MediaAssetResource asset, BlobServiceClient storage, CancellationToken cancellationToken)
+        private async Task<AnalysisResult> AnalyzeAsync<T>(T item, BlobServiceClient storage, CancellationToken cancellationToken)
         {
-            var result = new AnalysisResult(asset.Data.Name, MigrationStatus.NotMigrated);
-            _logger.LogDebug("Analyzing asset: {asset}, container: {container}", asset.Data.Name, asset.Data.Container);
+            string? itemName = null;
+            string? containerName = null;
+            BlobContainerClient? container = null;
+            AsyncPageable<MediaAssetStreamingLocator>? locators = null;
+            MediaAssetStorageEncryptionFormat? format = null;
+            if (item is MediaAssetResource mediaAsset)
+            {
+                itemName = mediaAsset.Data.Name;
+                containerName = mediaAsset.Data.Container;
+                format = mediaAsset.Data.StorageEncryptionFormat;
+                container = storage.GetContainer(mediaAsset);
+                locators = mediaAsset.GetStreamingLocatorsAsync();
+            }
+            else if (item is BlobContainerItem bcItem)
+            {
+                itemName = storage.AccountName;
+                container = storage.GetBlobContainerClient(bcItem.Name);
+                containerName = container.Name;
+                format = MediaAssetStorageEncryptionFormat.None;//todo StorageEncryptionFormat; 
+            }
+            if (itemName == null || container == null)
+            {
+                _logger.LogDebug("Analyzing item or container is null.");
+                throw new NullReferenceException();
+            }
+
+            var result = new AnalysisResult(itemName, MigrationStatus.NotMigrated);
+            _logger.LogDebug("Analyzing asset: {asset}, container: {container}", itemName, containerName);
             try
             {
-                var container = storage.GetContainer(asset);
+             
                 if (!await container.ExistsAsync(cancellationToken))
```

Comment:
For Storage account, the container must already exist,  the caller can guarantee it,  a new check here is not needed at all. ( it was needed for AMS account,  because the container for an asset might be deleted by customer somehow).

Reply:
the code path here combined both AMS account and storage account check.